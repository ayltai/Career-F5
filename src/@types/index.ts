export type { Course, } from './Course';
export type { Experience, } from './Experience';
export type { Profession, } from './Profession';
export type { RouteItem, } from './RouteItem';
export type { Section, } from './Section';
export type { Segmentation, } from './Segmentation';
