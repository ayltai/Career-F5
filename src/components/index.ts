export { ApplyAction, } from './ApplyAction';
export { ApplyDialog, } from './ApplyDialog';
export { ContactAction, } from './ContactAction';
export { Footer, } from './Footer';
export { PaymentPlan, } from './PaymentPlan';
export { RangeIndicator, } from './RangeIndicator';
export { TopBar, } from './TopBar';
