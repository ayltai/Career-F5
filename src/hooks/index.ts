export { useAppDispatch, useAppSelector, } from './states';
