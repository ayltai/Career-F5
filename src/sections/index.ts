export { BannerSection, } from './BannerSection';
export { ContactSection, } from './ContactSection';
export { CourseSection, } from './CourseSection';
export { InstructorSection, } from './InstructorSection';
export { OverviewSection, } from './OverviewSection';
export { PaymentSection, } from './PaymentSection';
export { ProspectSection, } from './ProspectSection';
export { StructureSection, } from './StructureSection';
