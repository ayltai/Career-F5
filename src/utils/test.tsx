import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
