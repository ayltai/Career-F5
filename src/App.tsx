import { WhatsApp as WhatsAppIcon, } from '@mui/icons-material';
import { Button, CssBaseline, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Fab, Stack, Theme, ThemeProvider, Typography, } from '@mui/material';
import mixpanel from 'mixpanel-browser';
import React, { Fragment, useEffect, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { ApplyAction, ApplyDialog, ContactAction, Footer, TopBar, } from './components';
import { applicationDefinition, } from './data';
import { Home, } from './screens';
import { useAppSelector, } from './hooks';
import { AppTheme, } from './styles';
import { handleError, } from './utils';

const contactAction = <ContactAction />;

const handleCallToActionClick = () => {
    if (process.env.REACT_APP_WHATSAPP_PHONE_NUMBER) {
        const newWindow = window.open(`https://wa.me/${ process.env.REACT_APP_WHATSAPP_PHONE_NUMBER }`, '_blank', 'noopener,noreferrer');
        if (newWindow) newWindow.opener = null;
    }
};

export const App = () => {
    const language = useAppSelector(state => state.preferences.language);

    const [ applyOpen, setApplyOpen, ] = useState<boolean>(false);
    const [ errorOpen, setErrorOpen, ] = useState<boolean>(false);
    const [ appTheme,  setAppTheme,  ] = useState<Theme>(AppTheme(language));

    const { t, i18n, } = useTranslation();

    const handleApplyOpen = () => setApplyOpen(true);

    const handleApplyClose = () => setApplyOpen(false);

    const handleErrorClose = () => setErrorOpen(false);

    const handleSubmit = (data : Record<string, string>) => fetch('https://corsproxy.io/?https://docs.google.com/forms/u/0/d/e/1FAIpQLSf2J4dcklFWXhHhO6glY5zJa6T9zreNR5uP-j8k3Eu_m1jgzQ/formResponse', {
        method  : 'POST',
        headers : {
            'Content-Type' : 'application/x-www-form-urlencoded',
        },
        body    : new URLSearchParams(data).toString(),
    }).then(() => {
        setApplyOpen(false);

        return true;
    }).catch(e => {
        handleError(e);

        setErrorOpen(true);

        return false;
    });

    const applyAction = (
        <ApplyAction onClick={handleApplyOpen} />
    );

    useEffect(() => mixpanel.track('Home'), []);

    useEffect(() => {
        i18n.changeLanguage(language).catch(handleError);
        setAppTheme(AppTheme(language));
    }, [ language, ]);

    return (
        <Fragment>
            <CssBaseline />
            <ThemeProvider theme={appTheme}>
                <a id='home' />
                <TopBar
                    branding={
                        <Fragment>
                            <Typography
                                marginRight={2}
                                variant='h4'
                                fontFamily='"Lobster Two"'
                                fontWeight='bold'>
                                {t('branding')}
                            </Typography>
                            <img
                                width={40}
                                height={40}
                                alt={t<string>('branding')}
                                src='/safari-pinned-tab.svg' />
                        </Fragment>
                    }
                    menu={[
                        {
                            text  : t<string>('courses'),
                            route : '#courses',
                        },
                        {
                            text  : t<string>('contact'),
                            route : '#contact',
                        },
                        {
                            text  : t<string>('about'),
                            route : '#instructor',
                        },
                    ]}
                    callToAction={
                        <Stack
                            spacing={2}
                            direction='row'>
                            {applyAction}
                            {contactAction}
                        </Stack>
                    }
                />
                <Fab
                    sx={{
                        display  : {
                            sx : 'block',
                            md : 'none',
                        },
                        position : 'fixed',
                        right    : 32,
                        bottom   : 32,
                    }}
                    color='secondary'
                    onClick={handleCallToActionClick}>
                    <WhatsAppIcon sx={{
                        color : 'white',
                    }} />
                </Fab>
                <Home
                    applyAction={applyAction}
                    contactAction={contactAction} />
                <Footer />
                <ApplyDialog
                    open={applyOpen}
                    onClose={handleApplyClose}
                    title={t<string>('apply_title')}
                    description={t<string>('apply_description')}
                    definition={applicationDefinition}
                    onSubmit={handleSubmit} />
                <Dialog
                    open={errorOpen}
                    onClose={handleErrorClose}>
                    <DialogTitle>
                        {t<string>('error_title')}
                    </DialogTitle>
                    <DialogContent>
                        <DialogContentText>
                            {t<string>('error_apply_failed')}
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button onClick={handleErrorClose}>
                            {t<string>('action_close')}
                        </Button>
                    </DialogActions>
                </Dialog>
            </ThemeProvider>
        </Fragment>
    );
};
