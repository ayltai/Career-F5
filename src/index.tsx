import * as Sentry from '@sentry/react';
import mixpanel from 'mixpanel-browser';
import React, { StrictMode, } from 'react';
import ReactDOM from 'react-dom/client';
import { Provider, } from 'react-redux';
import { PersistGate, } from 'redux-persist/integration/react';
import '@fontsource/kreon/400.css';
import '@fontsource/kreon/700.css';
import '@fontsource/lobster-two/400.css';
import '@fontsource/noto-sans-tc/chinese-traditional-400.css';
import '@fontsource/noto-sans-tc/chinese-traditional-700.css';
import '@fontsource/noto-serif-tc/chinese-traditional-400.css';
import '@fontsource/noto-serif-tc/chinese-traditional-700.css';
import '@fontsource/sansita/400.css';
import '@fontsource/sansita/400-italic.css';
import '@fontsource/sansita/700.css';

import PackageInfo from '../package.json';

import { App, } from './App';
import { applyI18n, } from './i18n';
import en from './i18n/en.json';
import zh from './i18n/zh.json';
import { persistor, store, } from './states';
import { handleError, } from './utils';
import './index.css';

if (process.env.REACT_APP_MIXPANEL_TOKEN) mixpanel.init(process.env.REACT_APP_MIXPANEL_TOKEN, {
    debug : process.env.NODE_ENV !== 'production',
});

if (process.env.REACT_APP_SENTRY_DSN) Sentry.init({
    dsn              : process.env.REACT_APP_SENTRY_DSN,
    environment      : process.env.NODE_ENV || 'development',
    release          : PackageInfo.version,
    tracesSampleRate : 1,
    integrations     : [
        new Sentry.BrowserTracing(),
    ],
});

applyI18n({
    language         : navigator.language.substring(0, 2),
    fallbackLanguage : 'en',
    resources        : {
        en : {
            translation : en,
        },
        zh : {
            translation : zh,
        },
    },
}).then(() => ReactDOM.createRoot(document.getElementById('root')!).render(
    <StrictMode>
        <Sentry.ErrorBoundary showDialog>
            <Provider store={store}>
                <PersistGate persistor={persistor}>
                    <App />
                </PersistGate>
            </Provider>
        </Sentry.ErrorBoundary>
    </StrictMode>
)).catch(handleError);
