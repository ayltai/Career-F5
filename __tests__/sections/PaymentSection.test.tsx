import React from 'react';

import { render, } from '../../src/utils/test';
import { PaymentSection, } from '../../src/sections';

describe('PaymentSection', () => {
    it('renders correctly', () => expect(render(
        <PaymentSection
            category='category'
            title='title'
            description='description' />
    )).toMatchSnapshot());
});
