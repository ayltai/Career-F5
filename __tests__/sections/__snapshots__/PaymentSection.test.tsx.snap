// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSection renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1nm38x2"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiStack-root css-1p5q5e5-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1ceq6vm-MuiTypography-root"
            >
              category
            </h6>
            <div
              class="MuiBox-root css-1qm1lh"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1vu909e-MuiTypography-root"
              >
                title
              </h3>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1fds5j6-MuiTypography-root"
            >
              description
            </p>
            <div
              class="MuiBox-root css-xuhmcu"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                    >
                      payment_full_title
                    </h5>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                        >
                          <div
                            class="MuiBox-root css-1sf3xto"
                          >
                            <h5
                              class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                            >
                              payment_full_cost
                            </h5>
                          </div>
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                          >
                            payment_full_description
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-xuhmcu"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                    >
                      payment_instalment_title
                    </h5>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                        >
                          <div
                            class="MuiBox-root css-1sf3xto"
                          >
                            <h5
                              class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                            >
                              payment_instalment_cost
                            </h5>
                          </div>
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                          >
                            payment_instalment_description
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-xuhmcu"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                    >
                      payment_discount_title
                    </h5>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                        >
                          <div
                            class="MuiBox-root css-1sf3xto"
                          >
                            <h5
                              class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                            >
                              payment_discount_cost
                            </h5>
                          </div>
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                          >
                            payment_discount_description
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-xuhmcu"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                    >
                      payment_interview_title
                    </h5>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                        >
                          <div
                            class="MuiBox-root css-1sf3xto"
                          >
                            <h5
                              class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                            >
                              payment_interview_cost
                            </h5>
                          </div>
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                          >
                            payment_interview_description
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1nm38x2"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1ceq6vm-MuiTypography-root"
          >
            category
          </h6>
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1vu909e-MuiTypography-root"
            >
              title
            </h3>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1fds5j6-MuiTypography-root"
          >
            description
          </p>
          <div
            class="MuiBox-root css-xuhmcu"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                  >
                    payment_full_title
                  </h5>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                      >
                        <div
                          class="MuiBox-root css-1sf3xto"
                        >
                          <h5
                            class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                          >
                            payment_full_cost
                          </h5>
                        </div>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                        >
                          payment_full_description
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-xuhmcu"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                  >
                    payment_instalment_title
                  </h5>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                      >
                        <div
                          class="MuiBox-root css-1sf3xto"
                        >
                          <h5
                            class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                          >
                            payment_instalment_cost
                          </h5>
                        </div>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                        >
                          payment_instalment_description
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-xuhmcu"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                  >
                    payment_discount_title
                  </h5>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                      >
                        <div
                          class="MuiBox-root css-1sf3xto"
                        >
                          <h5
                            class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                          >
                            payment_discount_cost
                          </h5>
                        </div>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                        >
                          payment_discount_description
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-xuhmcu"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
                  >
                    payment_interview_title
                  </h5>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                      >
                        <div
                          class="MuiBox-root css-1sf3xto"
                        >
                          <h5
                            class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                          >
                            payment_interview_cost
                          </h5>
                        </div>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                        >
                          payment_interview_description
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
